tests/lib/rules/no-await-sync-events.test.ts

```diff
@@ -148,31 +143,143 @@ ruleTester.run(RULE_NAME, rule, {
-    ...eventFunctions.map((func) => ({
+    // sync fireEvent methods with await operator are not valid
+    ...FIRE_EVENT_FUNCTIONS.map((func) => ({
       code: `
         import { fireEvent } from '@testing-library/framework';
+        test('should report fireEvent.${func} sync event awaited', async() => {
+          await fireEvent.${func}('foo');
+        });
+      `,
+      errors: [
+        {
+          line: 4,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: `fireEvent.${func}` },
+        },
+      ],
+    })),
+    // sync userEvent sync methods with await operator are not valid
+    ...USER_EVENT_SYNC_FUNCTIONS.map((func) => ({
+      code: `
         import userEvent from '@testing-library/user-event';
-        test('should report sync event awaited', async() => {
-          await ${func}('foo');
+        test('should report userEvent.${func} sync event awaited', async() => {
+          await userEvent.${func}('foo');
         });
       `,
-      errors: [{ line: 5, messageId: 'noAwaitSyncEvents' }],
+      errors: [
+        {
+          line: 4,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: `userEvent.${func}` },
+        },
+      ],
     })),
     {
       code: `
         import userEvent from '@testing-library/user-event';
-        test('should report sync event awaited', async() => {
-          await userEvent.type('foo', 'bar', {hello: 1234});
-          await userEvent.keyboard('foo', {hello: 1234});
+        test('should report async events without delay awaited', async() => {
+          await userEvent.type('foo', 'bar');
+          await userEvent.keyboard('foo');
+        });
+      `,
+      errors: [
+        {
+          line: 4,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: 'userEvent.type' },
+        },
+        {
+          line: 5,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: 'userEvent.keyboard' },
+        },
+      ],
+    },
+    {
+      code: `
+        import userEvent from '@testing-library/user-event';
+        test('should report async events with 0 delay awaited', async() => {
+          await userEvent.type('foo', 'bar', { delay: 0 });
+          await userEvent.keyboard('foo', { delay: 0 });
+        });
+      `,
+      errors: [
+        {
+          line: 4,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: 'userEvent.type' },
+        },
+        {
+          line: 5,
+          column: 17,
+          messageId: 'noAwaitSyncEvents',
+          data: { name: 'userEvent.keyboard' },
+        },
+      ],
+    },
+    {
+      settings: { 'testing-library/utils-module': 'test-utils' },
+      code: `
+        import { fireEvent as renamedFireEvent } from 'test-utils';
+        import renamedUserEvent from '@testing-library/user-event';
```

Comment:
Thanks to these 2 imports here I realized about it was required to save `user-event` imported module independently.